src/Core/Resolvers/DWSqlQueryBuilder.cs

```diff
@@ -133,11 +167,35 @@ private static string GenerateColumnsAsJson(SqlQueryStructure structure, bool su
                     stringAgg.Append($"N\'\"{escapedLabel}\":\' + ISNULL(\'\"\'+STRING_ESCAPE([{col_value}],'json')+\'\"\','null')");
                 }
 
-                i++;
+                columnCount++;
 
                 // Add comma if not last column. example: {"id":"1234","name":"Big Company"}
                 // the below ensures there is a comma after id but not after name.
-                if (i != structure.Columns.Count)
+                if (columnCount != structure.Columns.Count)
+                {
+                    stringAgg.Append("+\',\'+");
+                }
+            }
+
+            int aggregationColumnCount = 0;
+            // Handle aggregation columns
+            foreach (AggregationOperation aggregation in structure.GroupByMetadata.Aggregations)
+            {
+                if (aggregationColumnCount == 0 && columnCount != 0)
+                {
+                    // need to add a comma if there are columns before the aggregation columns
+                    stringAgg.Append("+\',\'+");
+                }
+
+                string col_value = aggregation.Column.OperationAlias;
+                col_value = $"CONVERT(NVARCHAR(MAX), [{col_value}])";
+                string escapedLabel = aggregation.Column.OperationAlias.Replace("'", "''");
+
+                stringAgg.Append($"N\'\"{escapedLabel}\":\' + ISNULL(STRING_ESCAPE({col_value},'json'),'null')");
```

Comment:
I notice the creation of Json is a common pattern here: ISNULL(STRING_ESCAPE(col_value), json) and on lines 156,161,167 - opportunity to create a common function.